Event Planning and Collaboration Tool/src/main/java/bg/sofia/uni/fmi/web/project/service/BudgetService.java

```diff
@@ -0,0 +1,152 @@
+package bg.sofia.uni.fmi.web.project.service;
+
+import bg.sofia.uni.fmi.web.project.dto.BudgetDto;
+import bg.sofia.uni.fmi.web.project.enums.ExpenditureCategory;
+import bg.sofia.uni.fmi.web.project.model.Budget;
+import bg.sofia.uni.fmi.web.project.model.Event;
+import bg.sofia.uni.fmi.web.project.repository.BudgetRepository;
+import bg.sofia.uni.fmi.web.project.validation.ResourceNotFoundException;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Positive;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class BudgetService {
+    private final BudgetRepository budgetRepository;
+
+    @Autowired
+    public BudgetService(BudgetRepository budgetRepository) {
+        this.budgetRepository = budgetRepository;
+    }
+
+    public Budget createBudget(
+        @NotNull(message = "Expense cannot be null")
+        Budget budgetToSave) {
+        budgetToSave.setCreationTime(LocalDateTime.now());
```

Comment:
Може би тези да идат във Facade класа

Reply:
When creating objects, for consistency reasons, I've left the setting of the creation time in the main service. The method createBudget() is public, so if for some reason, budget gets created from there, it will still have a creation time and deleted property.

Reply:
Yes, we have talked it